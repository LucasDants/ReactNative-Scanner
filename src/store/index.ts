import { IQRCodeList } from './modules/qrlist/types'
import rootReducer from './modules/rootReducer'
import { configureStore } from '@reduxjs/toolkit'

export interface IState {
    qrlist: IQRCodeList
}

const store = configureStore({reducer: rootReducer}) 

export default store