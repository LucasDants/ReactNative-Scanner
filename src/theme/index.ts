export default {
  colors: {
    red: {
      900: '#B83341',
      800: '#E03F50',
    },
    blue: '#0000FF',
    white: '#FFFFFF',
    gray: {
      100: '#DCDCDC',
      600: '#404040',
      900: '#030303',
    },
  },
};