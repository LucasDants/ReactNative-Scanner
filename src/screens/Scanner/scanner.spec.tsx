import React from 'react'

import { render } from '@testing-library/react-native'

import { ThemeProvider } from 'styled-components/native'
import theme from '../../theme'

import { Scanner } from '.'
import { Provider } from 'react-redux'
import store from '../../store'

const Providers = ({ children }) => (
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      { children }
    </ThemeProvider>
  </Provider>
)

describe('Scanner Screen', () => {

  it("should match the snapshot test", () => {
    const { toJSON } = render(
      <Scanner />, {wrapper: Providers}
    )

    const component = toJSON()
    expect(component).toMatchSnapshot()

  })
})