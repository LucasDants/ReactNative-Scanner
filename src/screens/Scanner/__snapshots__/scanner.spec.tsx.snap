// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scanner Screen should match the snapshot test 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#030303",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4290261825,
          4292886352,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomColor": "#B83341",
          "borderBottomWidth": 1,
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 33,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 51,
          },
        ]
      }
    >
      Scanner
    </Text>
  </ViewManagerAdapter_ExpoLinearGradient>
  <ViewManagerAdapter_ExpoBarCodeScannerView
    proxiedProperties={
      Object {
        "barCodeTypes": Array [],
        "onBarCodeScanned": [Function],
        "type": undefined,
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
